### PR TITLE
Candidate fix for SD too short

### PR DESCRIPTION
**Pull request by msettles**
ignored read if read is too short to contain a key, goes into ignored pile

### CODE REVIEW

**Comment by dstreett** [comment]
Is there a reason we should ignore reads versus just exiting? I'm happy either way, but the user did input invalid data for this algorithm, so I feel an exit is most appropriate?

**Comment by msettles** [comment]
Exiting would prevent certain workflows when only 1 or 2 reads go wonky. I
wouldn't call this invalid, but rather prior processing creates reads that
impact algorithm assumptions. In SD we also ignore all reads with a N char
in the key section,  I would consider this similar to that. Unable to SD
the read so ignore.

On Jul 27, 2018 2:31 PM, "David Streett" <notifications@github.com> wrote:

Is there a reason we should ignore reads versus just exiting? I'm happy
either way, but the user did input invalid data for this algorithm, so I
feel an exit is most appropriate?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ibest/HTStream/pull/141#issuecomment-408543836>, or mute
the thread

File: common/src/read.cpp [hunk 1]
```diff
@@ -32,9 +32,13 @@ std::string Read::subseq(size_t _start, size_t _length){
 }
 
 //PairedEndRead
-boost::optional<BitSet> PairedEndRead::get_key(size_t start, size_t length){
-    return std::max(str_to_bit(one.subseq(start, length) + two.subseq(start, length)),
-                    str_to_bit(two.subseq(start, length) + one.subseq(start, length)));
+boost::optional<BitSet> PairedEndRead::get_key(size_t _start, size_t _length){
+    if (one.getLength() <= (_start+_length) | two.getLength() <= (_start+_length)){
```
Comment (dstreett) [inline common/src/read.cpp]: Isn't this a bitwise or?

Comment (dstreett) [reply common/src/read.cpp]: Also, I'd prefer `std::max(one.getLength(), two.getLength()) <= _start+_length` 

And, last thing. Any reason why equal to would mess things up?

**Pull request merged**